src/commands/react.ts

```diff
@@ -0,0 +1,168 @@
+
+    // Args should only be length 1 or 2
+    if (commandArgs.length !== 1 && commandArgs.length !== 2) {
+      // Error, number of arguments doesn't match
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  You must either include an emoji name or a message-id and then an emoji name!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Argument error occurred', result: {success: false}};
+    }
+
+    if (commandArgs.length === 1) {
+      return await this.reactNow(commandArgs[0], message);
+    } else {
+      return await this.reactToMessage(commandArgs[0], commandArgs[1], message);
+    }
+  }
+
+  /**
+   * React to the message immediately above
+   * @param emojiName - The emoji name
+   * @param message - The command message
+   */
+  async reactNow(emojiName: string, message: Discord.Message): Promise<ICommandResult> {
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
+    }
+
+    let fetchedMessage: Discord.Collection<string, Message> = await message.channel
+      .fetchMessages({limit: 1, before: message.id});
+    if (!fetchedMessage || !(fetchedMessage instanceof Discord.Collection) ||
+      !(fetchedMessage.array()) || fetchedMessage.array().length < 1) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find a message to react to!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not find target message', result: {success: false}};
+    }
+
+    const targetMessage = fetchedMessage.array()[0];
+
+    const success = await this.reactService.tempReactToMessage(targetMessage, emoji, message);
+
+    if (success === false) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not apply the reaction!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not apply the reaction', result: {success: false}};
+    }
+
+    return {};
+  }
+
+  /**
+   * React to a specific message given by its Id
+   * @param messageId - The Id of the specific message
+   * @param emojiName - The name of the emoji
+   * @param message - The command message
+   */
+  async reactToMessage(messageId: string, emojiName: string, message: Discord.Message): Promise<ICommandResult> {
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
```

Comment:
Same as above. This is copy pasted a lot so it definitely be refactored into the above function.